controller/kubernetes/kubernetes.go

```diff
@@ -524,6 +565,89 @@ func (lbc *loadBalancerController) GetLBConfigs() ([]*config.LoadBalancerConfig,
 	return lbConfigs, nil
 }
 
+func (lbc *loadBalancerController) getProbeConfig(probeStr string, frontEndServices []*config.FrontendService, backendUUIDSvcMap map[string]string) (string, error) {
+	var probeConfig string
+
+	if probeStr != "" {
+		dec := json.NewDecoder(strings.NewReader(probeStr))
+		probeStructs := &ingressHealthChecks{}
+		err := dec.Decode(&probeStructs)
+		if err != nil {
+			return "", fmt.Errorf("Error unmarshaling k8s livenessProbes: %v err: %v", probeStr, err)
+		}
+		if probeStructs != nil {
```

Comment:
Nit: for better readability, can we return/continue when something is nil? like:

if probeStruct == nil {
return
}

Same on the lin e582